spec/services/secure_headers_whitelister_spec.rb

```diff
@@ -15,4 +15,25 @@ def extract_domain(url)
       end
     end
   end
+
+  describe '.csp_with_sp_redirect_uris' do
+    def csp_with_sp_redirect_uris(domain, sp_redirect_uris)
+      SecureHeadersWhitelister.csp_with_sp_redirect_uris(domain, sp_redirect_uris)
+    end
+
+    it 'generates the proper CSP array from action_url domain and ServiceProvider#redirect_uris' do
+      aggregate_failures do
+        domain = 'example1.com'
+        test_sp_uris = ['x-example-app://test', 'https://example2.com']
+        full_return = ["'self'", 'example1.com', 'x-example-app://test', 'https://example2.com']
+
+        expect(csp_with_sp_redirect_uris(domain, test_sp_uris)).to eq(full_return)
+
+        expect(csp_with_sp_redirect_uris(domain, test_sp_uris[0..0])).to eq(full_return[0...3])
+
+        expect(csp_with_sp_redirect_uris(domain, [])).to eq(full_return[0...2])
+        expect(csp_with_sp_redirect_uris(domain, nil)).to eq(full_return[0...2])
```

Comment:
The range here confused me at first until I noticed it was using the 3 dot version. Was there a particular reason you chose `[0...2]` vs `[0..1`]?

Reply:
No good reason, will change so it's clearer